### PR TITLE
#162949009 Enable user to highlight text and comment

### PR DESCRIPTION
**Pull request by charitymarani**
**What does this PR do?**
Users can highlight and comment on text

**Endpoints involved:**
```
/api/v1/articles/<slug>/highlight/

```
Adds tests for highlight and comment functionality

**How to manually test these features**
Clone the Repo
Make Migrations
Test the endpoints on postman

**Description of tasks to be completed**
As a user i should be able to highlight any text in an article(created by me or not) and comment on it.

**Relevant pivotal tracker story**
[#162949009](https://www.pivotaltracker.com/story/show/162949009)

**Screenshots**

**Highlight and comment on text**

screenshot 2019-01-31 at 09 14 04

**Get comment on highlight**

screenshot 2019-01-31 at 09 21 46

**Hightlight non existing article**

screenshot 2019-01-31 at 09 16 53

**Highlight indices that are not integers**

screenshot 2019-01-31 at 09 17 33

**Highlight index out of range**

screenshot 2019-01-31 at 10 13 53

**Start index greater than end index**

screenshot 2019-01-31 at 09 25 49

### CODE REVIEW

**Review by eduhmik — COMMENTED** [review]
Hey @charitymarani, keep up for the good work you've done... However in your screenshots highlight an index out of range, kindly show us the payload you are passing to get that response

**Comment by charitymarani** [comment]
> Hey @charitymarani, keep up for the good work you've done... However in your screenshots highlight an index out of range, kindly show us the payload you are passing to get that response

Done

File: authors/apps/articles/tests/test_highlight.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def test_missing_required_field(self):
+        """Test for missing field."""
+        message = "start_index field cannot be empty"
+        response = self.highlight()[4]
+        self._400_bad_request(response, message)
+
+    def test_highlight_non_existing_article(self):
+        """Test highlighting non-existing article"""
+        message = "No article found for the slug given"
+        response = self.highlight()[5]
```
Comment (Nicanor008) [inline authors/apps/articles/tests/test_highlight.py:70]: Does [5] represent the number of highlighted text, or??

Comment (charitymarani) [reply authors/apps/articles/tests/test_highlight.py:70]: I have a function called highlight()  that returns a list of different responses.5 is the index of the response in that list

Comment (Nicanor008) [reply authors/apps/articles/tests/test_highlight.py:70]: so the responses in the list are of different formats i.e. successful, unsuccessful, unavailable etc. Got it. That was simple to implement in a single function indeed.

File: authors/apps/articles/tests/test_highlight.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        response5 = self.client.post(self.highlight_url(
+            slug), self.missing_highlight_field, format='json')
+        response6 = self.client.post(self.highlight_url(
+            "invalid slug"), self.highlight_data, format='json')
+        responses = [response1, response2, response3, response4,
+                     response5, response6]
+        return responses
+
+    def base_error_messages(self, response, message):
+        self.assertIn(message.encode(), response.content)
```
Comment (JoshuaKodhe) [inline authors/apps/articles/tests/test_highlight.py:30]: I like the use of the function as it is reusable. 
Keep up the good use of reusability.

**Pull request merged**